web-app/src/app/(app)/dashboard/gallery/components/filter-section.tsx

```diff
@@ -0,0 +1,75 @@
+"use client";
+
+import { X } from "lucide-react";
+import { usePathname, useRouter, useSearchParams } from "next/navigation";
+import { useTranslations } from "next-intl";
+import { useCallback, useEffect, useState } from "react";
+
+import { Button } from "@/components/ui/button";
+import { Input } from "@/components/ui/input";
+
+import Tags from "./tags";
+
+export default function FilterSection() {
+  const t = useTranslations("App.Gallery.FilterSection");
+
+  const { replace } = useRouter();
+  const pathname = usePathname();
+  const searchParams = useSearchParams();
+
+  const [query, setQuery] = useState<string>(searchParams.get("query") ?? "");
+  const [selectedTags, setSelectedTags] = useState<string[]>(
+    searchParams.get("tags")?.split(",") ?? [],
+  );
+
+  const handleSearch = useCallback(
+    (query: string, tags: string[]) => {
+      const params = new URLSearchParams(searchParams);
+      if (query) params.set("query", query);
+      else params.delete("query");
+      if (tags.length > 0) params.set("tags", tags.join(","));
+      else params.delete("tags");
+      replace(`${pathname}?${params.toString()}`);
+    },
+    [pathname, searchParams, replace],
+  );
+
+  const handleReset = () => {
+    setQuery("");
+    setSelectedTags([]);
+    replace(pathname);
+  };
+
+  useEffect(() => {
+    const timerId = setTimeout(() => {
+      handleSearch(query, selectedTags);
+    }, 500);
+    return () => clearTimeout(timerId);
+  }, [query, selectedTags, handleSearch]);
```

Comment:
I recommend using [`use-debounce`](https://github.com/xnimorz/use-debounce) for debouncing. It's very elegant.